.circleci/config.yml

```diff
@@ -109,6 +109,49 @@ jobs:
             cargo install --force cargo-audit
             cargo audit
       - build_teardown
+  code-coverage:
+    description: Run code coverage
+    machine:
+      image: ubuntu-1604:201903-01
+    resource_class: large
+    steps:
+      - checkout
+      - run:
+          name: Install deps
+          command: |
+            sudo apt-get update
+            sudo apt-get install -y protobuf-compiler cmake curl clang
+            curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y --default-toolchain none
+      - run:
+          name: Setup PATH
+          command: echo "export PATH=~/.cargo/bin:$PATH" >> $BASH_ENV
+      - env_setup
+      - run:
+          name: Install nightly toolchain (profile feature is not yet in beta/stable)
+          command: rustup install nightly
+      - run:
+          name: Install grcov
+          command: cargo install --force grcov
+      - run:
+          name: Setup rustc flags
+          command: |
+            echo "export CARGO_INCREMENTAL=0" >> $BASH_ENV
+            echo "export RUSTFLAGS='-Zprofile -Ccodegen-units=1 -Cinline-threshold=0 -Clink-dead-code -Coverflow-checks=off -Zno-landing-pads'" >> $BASH_ENV
+      - run:
```

Comment:
Is it necessary to first do a "cargo build"? The subsequent "cargo test" will rebuild at least most of the code anyway (AFAIK) and the "cargo build" will compile some code that cannot be exercised in the tests. Unless there's a good reason, I think you should skip this.

Reply:
Good point. The build step generates additional artifacts we need to exclude or they end up in the report.  Will patch this up.